backend/dataall/db/api/target_type.py

```diff
@@ -1,64 +1,43 @@
-            environment=(permissions.GET_ENVIRONMENT, permissions.UPDATE_ENVIRONMENT),
-            notebook=(permissions.GET_NOTEBOOK, permissions.UPDATE_NOTEBOOK),
-            mlstudio=(
-                permissions.GET_SGMSTUDIO_NOTEBOOK,
-                permissions.UPDATE_SGMSTUDIO_NOTEBOOK,
-            ),
-            pipeline=(permissions.GET_PIPELINE, permissions.UPDATE_PIPELINE),
-            redshift=(
-                permissions.GET_REDSHIFT_CLUSTER,
-                permissions.GET_REDSHIFT_CLUSTER,
-            ),
-        )
+    """Resolves the read/write permissions for different type of resources (target types)"""
+    _TARGET_TYPES = {}
+
+    def __init__(self, name, read_permission, write_permission):
+        self.name = name
+        self.read_permission = read_permission
+        self.write_permission = write_permission
+
+        TargetType._TARGET_TYPES[name] = self
 
     @staticmethod
     def get_resource_update_permission_name(target_type):
         TargetType.is_supported_target_type(target_type)
-        return TargetType.get_target_type_permissions()[target_type][1]
+        return TargetType._TARGET_TYPES[target_type].write_permission
 
     @staticmethod
     def get_resource_read_permission_name(target_type):
         TargetType.is_supported_target_type(target_type)
-        return TargetType.get_target_type_permissions()[target_type][0]
+        return TargetType._TARGET_TYPES[target_type].read_permission
 
     @staticmethod
     def is_supported_target_type(target_type):
-        supported_types = [
-            'dataset',
-            'environment',
-            'notebook',
-            'mlstudio',
-            'pipeline',
-            'redshift',
-        ]
-        if target_type not in supported_types:
+        if target_type not in TargetType._TARGET_TYPES:
             raise exceptions.InvalidInput(
                 'targetType',
                 target_type,
-                ' or '.join(supported_types),
+                ' or '.join(TargetType._TARGET_TYPES.keys()),
             )
 
-    @staticmethod
-    def get_target_type(model_name):
-        target_types_map = dict(
-            environment=models.Environment,
-            dataset=models.Dataset,
-            notebook=models.SagemakerNotebook,
-            mlstudio=models.SagemakerStudioUserProfile,
-            pipeline=models.DataPipeline,
-            redshift=models.RedshiftCluster,
-        )
-        return [k for k, v in target_types_map.items() if v == model_name][0]
+
+TargetType("dataset", permissions.GET_DATASET, permissions.UPDATE_DATASET)
+TargetType("environment", permissions.GET_ENVIRONMENT, permissions.UPDATE_ENVIRONMENT)
+TargetType("mlstudio", permissions.GET_SGMSTUDIO_NOTEBOOK, permissions.UPDATE_SGMSTUDIO_NOTEBOOK)
+TargetType("pipeline", permissions.GET_PIPELINE, permissions.UPDATE_PIPELINE)
+TargetType("redshift", permissions.GET_REDSHIFT_CLUSTER, permissions.GET_REDSHIFT_CLUSTER)
```

Comment:
This will go to the modules once we create them right?

Reply:
Yes, it's correct!